V5.28 documentation (#3996)
mattermost/docs: source/administration/changelog.md

Also see [changelog in progress](http://bit.ly/2nK3cVf) for the next release.

## Release v5.28 - [Feature Release](https://docs.mattermost.com/process/release-faq.html#release-overview)

**Release day: 2020-10-16**

### Compatibility

>  - PostgreSQL ended long-term support for [version 9.4 in February 2020](https://www.postgresql.org/support/versioning). Mattermost is officially supporting PostgreSQL version 10 with v5.26 release as PostgreSQL 9.4 is no longer supported. New installs will require PostgreSQL 10+. Previous Mattermost versions, including our current ESR, will continue to be compatible with PostgreSQL 9.4. We plan on fully deprecating PostgreSQL 9.4 In our v5.30 release (December 16). Please follow the instructions under the Upgrading Section within [the PostgreSQL documentation](https://www.postgresql.org/support/versioning/).

Reviewer: ```suggestion
 - PostgreSQL ended long-term support for [version 9.4 in February 2020](https://www.postgresql.org/support/versioning). Mattermost is officially supporting PostgreSQL version 10 from v5.26 release as PostgreSQL 9.4 is no longer supported. New installs will require PostgreSQL 10+. Previous Mattermost versions, including our current ESR, will continue to be compatible with PostgreSQL 9.4. We plan on fully deprecating PostgreSQL 9.4 In our v5.30 release (December 16 2020). Please follow the instructions under the Upgrading Section within [the PostgreSQL documentation](https://www.postgresql.org/support/versioning/).
```

>  - Support for server [Extended Support Release](https://docs.mattermost.com/administration/extended-support-release.html) (ESR) 5.19 has come to the end of its lifecycle. Upgrading to server v5.25 or later is required.

Reviewer: ```suggestion
 - Support for Mattermost Server [Extended Support Release](https://docs.mattermost.com/administration/extended-support-release.html) (ESR) 5.19 has come to the end of its lifecycle. Upgrading to Mattermost Server v5.25 or later is required.
```

>  - TLS versions 1.0 and 1.1 have been deprecated by browser vendors. Starting in v5.31 (January 16th) mmctl will return an error when connected to Mattermost servers deployed with these TLS versions and System Admins will need to explicitly add a flag in their commands to continue to use them. We recommend upgrading to TLS version 1.2 or higher.

Reviewer: ```suggestion
 - TLS versions 1.0 and 1.1 have been deprecated by browser vendors. Starting in v5.31 (January 16 2021) mmctl will return an error when connected to Mattermost servers deployed with these TLS versions and System Admins will need to explicitly add a flag in their commands to continue to use them. We recommend upgrading to TLS version 1.2 or higher.
```

### Breaking Changes
 - Now when the service crashes, it will generate a coredump instead of just dumping the stack trace to the console. This allows us to preserve the full information of the crash to help with debugging it. For more information about coredumps, please see: https://man7.org/linux/man-pages/man5/core.5.html.  

> **IMPORTANT:** If you upgrade from a release earlier than 5.27, please read the other [Important Upgrade Notes](https://docs.mattermost.com/administration/important-upgrade-notes.html).

Reviewer: ```suggestion
**IMPORTANT:** If you upgrade from a release earlier than v5.27, please read the other [Important Upgrade Notes](https://docs.mattermost.com/administration/important-upgrade-notes.html).
```

### Highlights

#### New admin roles to delegate administration tasks to other types of administrators (E20)
 - New admin roles are additional system roles that have access to designated areas of the System Console. This enables you to delegate certain administrative tasks to other members of your organization.

#### Certificate-based authentication with AD/LDAP (E10)
 - You can now improve the security of your AD/LDAP authentication with certificate-based AD/LDAP authentication.

#### Stay current with in-product notices

>  - Now with in-product notices, users and Admins will be made aware of the newest product enhancements from within Mattermost. [Learn more about in-product notices here](https://about.mattermost.com/default-notices/).

Reviewer: ```suggestion
 - With in-product notices, users and Admins will be made aware of the newest product enhancements from within Mattermost. [Learn more about in-product notices here](https://about.mattermost.com/default-notices/).
```

 
#### Plugins
 - Plugins now start concurrently on server startup.
 - Plugin tooltips are now only rendered when user hovers over a link.
 - Added a ``CreateCommand`` plugin API that creates a slash command that is not handled by the plugin itself.
 
#### Administration
 - Added the ability to upload and remove private and public certicates for LDAP authentication.
 - Added support for resumable file uploads.
 - Added the ability to convert a public channel to private and vice versa via Advanced Permissions.
 - Added filters to search teams in Teams page.
 - Improved logging related to sessions that are not found.
 - Created Grafana enterprise metrics for logging, such as for current queue level(s), rate of logging records emitted, and rate of logging errors.
 - Improved logging when ``GetUser`` fails during MFA Authentication.
 - Added support for sending telemetry via an environment variable set by packages to identify type of deployment (e.g. Docker, Mattermost Omnibus).

### Bug Fixes

>  - Fixed an issue where a large amount of archived channels caused performance degredation.

Reviewer: ```suggestion
 - Fixed an issue where a large number of archived channels caused performance degradation.
```

 - Fixed an issue where users were unable to type color hex value into custom theme color input box.
 - Fixed an issue where the badge with a mention count on the team sidebar did not increment when user was added to a channel.
 - Fixed an issue where Group Message results were prioritized over Direct Message results for Full Name in the user autocomplete.
 - Fixed an issue where the New Message indicator was broken when a webhook owned by the user posted to a channel.
 - Fixed an issue where the active search bar was not vertically aligned with left edge of the right-hand side in tablet view.
 - Fixed an issue where there were two scrollbars showing in the channel switcher.
 - Fixed an issue where the "Start trial" message was unreadable in the System Console on dark theme on first load.
 - Fixed an issue on Firefox where pasting an image also added the file as text.
 - Fixed an issue where Python syntax highlighting handled ``"""`` strangely.
 - Fixed an issue where formatting around inline codes was missing.
 - Fixed an issue where ``GetPluginStatus`` didn't work in a non-cluster environment.

### config.json

> Multiple setting options were added to `config.json`. Below is a list of the additions and their default values on install. The settings can be modified in `config.json`, or the System Console when available.

Reviewer: ```suggestion
Multiple setting options were added to ``config.json``. Below is a list of the additions and their default values on install. The settings can be modified in ``config.json``, or the System Console when available.
```

> #### Changes to Team Edition and Enterprise Edition:

Reviewer: ```suggestion
#### Changes to Team Edition and Enterprise Edition
```

### Open Source Components
 - Added ``react-is`` and ``tinycolor2`` to https://github.com/mattermost/mattermost-webapp.
 - Removed ``@types/highlight.js``, ``@typescript-eslint/parser``, ``bootstrap-colorpicker``, and ``intl`` from https://github.com/mattermost/mattermost-webapp.
 - Removed ``react-native-v8`` from https://github.com/mattermost/mattermost-mobile.

### Database Changes
 - Added a new column ``Commands.PluginId``.
 - Changed to data type of ``Teams.Type to varchar(255)``.
 - Changed to data type of ``Teams.SchemeId to varchar(26)``.
 - Changed to data type of ``IncomingWebhooks.Username to varchar(255)``.
 - Changes to data type of ``IncomingWebhooks.IconURL to text",``.

### API Changes
 - Added ``POST /upgrade_to_enterprise`` API endpoint.
 - Added ``GET /upgrade_to_enterprise/status`` API endpoint.
 - Added ``POST /restart`` API endpoint.
 - Added ``GET /warn_metrics/status`` API endpoint.
 - Added ``POST /warn_metrics/ack/:warn_metric_id`` API endpoint.
 
### Known Issues
 - Emoji counter in the center channel doesn't always update immediately when a reaction is added in the right-hand side.

>  - Pressing **Enter** closes the Account Settings Edit modal when adjusting the settings for desktop notification sound.

Reviewer: ```suggestion
 - Pressing ENTER closes the Account Settings Edit modal when adjusting the settings for desktop notification sound.
```
